File: javascript_basics/src/utils/logger.js
const winston = require("winston");
const fs = require("fs-extra");
const path = require("path");

const logsDir = path.join(process.cwd(), "results", "logs");
fs.ensureDirSync(logsDir);

const timestamp = new Date().toISOString().replace(/[:.]/g, "-");
const logFile = path.join(logsDir, `test_run_${timestamp}.log`);

const logger = winston.createLogger({
  level: "info",
  format: winston.format.combine(
    winston.format.timestamp({ format: "YYYY-MM-DD HH:mm:ss" }),
    winston.format.printf(({ timestamp, level, message }) => {
      return `[${timestamp}] ${level.toUpperCase()}: ${message}`;
    })
  ),
  transports: [
    new winston.transports.File({ filename: logFile }),
    new winston.transports.Console()
  ],
});

module.exports = { logger };

File: ruby_basics/src/support/logger.rb
require "logger"
require "fileutils"

module TestLogger
  ROOT_DIR = File.expand_path("../..", __dir__)            # => ruby_basics/
  LOGS_DIR = File.join(ROOT_DIR, "results", "logs")        # => ruby_basics/results/logs

  def self.create_logger
    FileUtils.mkdir_p(LOGS_DIR)

    timestamp = Time.now.strftime("%Y%m%d_%H%M%S")
    log_path  = File.join(LOGS_DIR, "test_run_#{timestamp}.log")

    logger = Logger.new(log_path)
    logger.level = Logger::INFO
    logger.datetime_format = "%Y-%m-%d %H:%M:%S"
    logger
  end
end
